### PR TITLE
Merge ClientIp with Headers.

### PR DESCRIPTION
**Pull request by BlackDex**
Since we now use the `ClientIp` Guard on a lot more places, it also increases the size of binary, and the macro generated code because of this extra Guard. By merging the `ClientIp` Guard with the several `Header` guards we have it reduces the amount of code generated (including LLVM IR), but also a small speedup in build time.

I also spotted some small `json!()` optimizations which also reduced the amount of code generated.

There was no decrease in the speed of the code as far as i could tell. It btw also lowered the amount of allocations done.

### CODE REVIEW

File: src/api/notifications.rs [hunk 1]
```diff
@@ -51,29 +49,6 @@ fn websockets_err() -> EmptyResult {
     }
 }
 
-#[post("/hub/negotiate")]
-fn negotiate(_headers: Headers) -> Json<JsonValue> {
```
Comment (dani-garcia) [inline src/api/notifications.rs:55]: Is this no longer used by the clients?

Comment (BlackDex) [reply src/api/notifications.rs:55]: I can't find that anywhere in the client/mobile code at all.
I could be wrong.

Comment (dani-garcia) [reply src/api/notifications.rs:55]: They are using SignalR so there is a chance that the library is using it internally, but the ws connection works without it, maybe it's not required anymore.

Comment (BlackDex) [reply src/api/notifications.rs:55]: Yea, i would have expected it to turn up in the builded coded in the end

Comment (dani-garcia) [reply src/api/notifications.rs:55]: Just checked the clients and they don't seem to be making the request so it's probably fine

**Review by dani-garcia — APPROVED** [review]

File: src/api/notifications.rs (hunk 1 above)
Comment (jjlin) [reply src/api/notifications.rs:55]: * https://github.com/bitwarden/clients/commit/31a257407be7f8f47624b0d021363aaf2cfda2d7
* https://learn.microsoft.com/en-us/aspnet/core/signalr/configuration?view=aspnetcore-7.0&tabs=javascript

Comment (BlackDex) [reply src/api/notifications.rs:55]: Thanks! I couldn't find that part. But of course i should have looked for a different spelling.

**Pull request merged**